src/utils/hooks/useGetResponsesAndElementState.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect } from 'react'
+import { useGetElementsAndResponsesQuery, GetElementsAndResponsesQuery } from '../generated/graphql'
+import {
+  ResponsesByCode,
+  ResponsesFullByCode,
+  TemplateElementState,
+  EvaluatedElement,
+} from '../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+
+const useGetResponsesAndElementState = (props: { serialNumber: string }) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [responsesFullByCode, setResponsesFullByCode] = useState({})
+  const [elementsExpressions, setElementsExpressions] = useState([{}])
+  const [elementsState, setElementsState] = useState({})
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(true)
+  const { data, loading: apolloLoading, error: apolloError } = useGetElementsAndResponsesQuery({
+    variables: { serial: serialNumber },
+  })
```

Comment:
There is a problem with that, I think. For the cached data to be used it would check by id, and since we aren't passing the id here it is just making another call to the server. I think we can make a different issue to solve that, but just flagging this here. That's why I was storing the Application `id` in the context area, and we would need to use the skip to not call this hook until we have the application loaded. Otherwise it's just making 2 calls to the server.

Reply:
Yes, I was hoping to use the id, but I knew it wouldn't have finished loading yet. @andreievg  has a couple of ideas of how to make one hook wait for the preceding one (he wrote a comment on it somewhere, but can't find right now).

We should optimise this once we've got some of the other things solidly in place.

Reply:
I think this is what you are looking for: https://github.com/openmsupply/application-manager-web-app/compare/example-skip-on-useQuery

By the way - I'm thinking about changing the cache for `Application` to use the serial as the identifier for applications stored in the cache. I've done this already in the branch that I'm currently working - but I'm still seeing calls to the graphql server when this runs so I'll continue to investigate. 🕵️‍♀️

Reply:
Yes, it's sorted now on the PR to-be-created. I fixed by using the `serial` as the primary key on the GraphQL queries (no changes on the back-end). And also changing the order and fields queried in each query.